File: visual/AssetsDescriptor/elephantHead.ts
import { FrameByState } from "."
import { UnitState } from "logic-contants"

const frameByState: FrameByState = {
  [UnitState.RUN]: {
    prefix: 'elephant_head_run',
    animationLength: 16,
    angles: 12,
    timePerFrame: 50, // speed in seconds
    frames: [],
  },
  [UnitState.SHOOT]: {
    prefix: 'elephant_head_shoot',
    animationLength: 6,
    angles: 16,
    timePerFrame: 60,
    frames: [],
  },
  [UnitState.ABILITY]: {
    prefix: 'elephant_head_shoot',
    animationLength: 6,
    angles: 16,
    timePerFrame: 60,
    frames: [],
  },
  [UnitState.CHASING]: {
    prefix: 'elephant_head_shoot',
    animationLength: 6,
    angles: 16,
    timePerFrame: 60,
    frames: [],
  },
  [UnitState.DIE]: {
    prefix: 'elephant_head_shoot',
    animationLength: 6,
    angles: 16,
    timePerFrame: 60,
    frames: [],
  },
  [UnitState.FLY]: {
    prefix: 'elephant_head_shoot',
    animationLength: 6,
    angles: 16,
    timePerFrame: 60,
    frames: [],
  },
  [UnitState.GETUP]: {
    prefix: 'elephant_head_shoot',
    animationLength: 6,
    angles: 16,
    timePerFrame: 60,
    frames: [],
  },
  [UnitState.IDLE]: {
    prefix: 'elephant_head_shoot',
    animationLength: 6,
    angles: 16,
    timePerFrame: 60,
    frames: [],
  }
}

export default frameByState

File: crate/src/vertex/assets_descriptor/elephant_head.rs
use std::collections::HashMap;

use crate::constants::State;

use super::AnimationDetails;

pub fn get() -> HashMap<State, AnimationDetails> {
    HashMap::from([
        (
            State::RUN,
            AnimationDetails {
                prefix: "elephant_head_run".to_owned(),
                animation_length: 16,
                angles: 12,
                time_per_frame: 50.0, // speed in seconds
                frames: vec![],
            },
        ),
        (
            State::SHOOT,
            AnimationDetails {
                prefix: "elephant_head_shoot".to_owned(),
                animation_length: 7,
                angles: 16,
                time_per_frame: 60.0,
                frames: vec![],
            },
        ),
        (
            State::IDLE,
            AnimationDetails {
                prefix: "elephant_head_idle".to_owned(),
                animation_length: 1,
                angles: 12,
                time_per_frame: 60.0,
                frames: vec![],
            },
        ),
    ])
}
